src/components/Footer/Footer.js

```diff
@@ -28,8 +39,21 @@ const Footer = ({
   blockStatus,
   loadingVote,
   showScoreButton = false,
-  onPressScore
+  onPressScore,
+  isShowDM = false
 }) => {
+  const {sendMessageDM} = useDMMessage();
+  const [profile] = React.useContext(Context).profile;
+  const [userAllowDm, setUserAllowDm] = React.useState(false);
+
+  const [loading, setLoading] = React.useState({
+    loadingDm: false,
+    loadingDmAnon: false,
+    loadingGetAllowAnonDmStatus: false
+  });
+  const refSheet = React.useRef();
+  const {createSignChat} = useCreateChat();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of local state for managing loading states is a good practice. However, it's better to create a separate function to handle the state updates to avoid repeating the same code in different places. This will make your code more maintainable and easier to read.

```diff
- setLoading({...loading, loadingDm: true});
+ setLoadingState('loadingDm', true);

// Add this function
const setLoadingState = (key, value) => {
  setLoading(prevState => ({ ...prevState, [key]: value }));
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->